test/extensions/transport_sockets/tls/cert_validator/default_validator_integration_test.cc

```diff
@@ -74,9 +74,42 @@ TEST_P(SslCertValidatorIntegrationTest, CertValidationFailed) {
   initialize();
   auto conn = makeSslClientConnection({});
   IntegrationCodecClientPtr codec = makeRawHttpConnection(std::move(conn), absl::nullopt);
+  ASSERT_TRUE(codec->connected());
+  test_server_->waitForCounterGe(listenerStatPrefix("ssl.handshake"), 1);
+  EXPECT_EQ(test_server_->counter(listenerStatPrefix("ssl.fail_verify_error"))->value(), 0);
+  codec->close();
+}
+
+// With only root trusted, certificate validation succeeds without max depth
+TEST_P(SslCertValidatorIntegrationTest, CertValidationSucceedWithTrustRootOnly) {
+  config_helper_.addSslConfig(ConfigHelper::ServerSslOptions()
+                                  .setRsaCert(true)
+                                  .setTlsV13(true)
+                                  .setClientWithIntermediateCert(true)
+                                  .setTrustRootOnly(true));
+  initialize();
+  auto conn = makeSslClientConnection({});
+  IntegrationCodecClientPtr codec = makeRawHttpConnection(std::move(conn), absl::nullopt);
+  ASSERT_TRUE(codec->connected());
+  test_server_->waitForCounterGe(listenerStatPrefix("ssl.handshake"), 1);
+  EXPECT_EQ(test_server_->counter(listenerStatPrefix("ssl.fail_verify_error"))->value(), 0);
+  codec->close();
+}
+
+// With only root ca trusted, certificate validation is expected to fail with max depth,
```

Comment:
could you add more comments about the server/client cert chain? that help people read this test in the future.

Reply:
done